File: javalibrary/src/main/java/com/bluthlee/javalibrary/FloatAudioRecorder.java
package com.bluthlee.javalibrary;

import android.content.Context;

import com.bluthlee.javalibrary.data.BaseBuilder;
import com.bluthlee.javalibrary.data.DataBinder;

/**
 * Created by LC on 2017/8/31.
 */

public class FloatAudioRecorder extends AudioRecorder {

    public static class Builder extends BaseBuilder {
        private Builder() {
        }

        private Context context;

        public static Builder setContext(Context context) {
            Builder builder = new Builder();
            builder.context = context;
            return builder;
        }

        @Override
        public Builder setOnFinishListener(OnFinishListener onFinishListener) {
            super.setOnFinishListener(onFinishListener);
            return this;
        }

        @Override
        public Builder setFileName(String fileName) {
            super.setFileName(fileName);
            return this;
        }

        @Override
        public Builder setDirPath(String dirPath) {
            super.setDirPath(dirPath);
            return this;
        }

        public FloatAudioRecorder build() {
            FloatAudioRecorder floatAudioRecorder = new FloatAudioRecorder();
            floatAudioRecorder.setStyle(DataBinder.STYLE_FLOAT);
            floatAudioRecorder.setListener(onFinishListener);
            floatAudioRecorder.setFileName(fileName);
            floatAudioRecorder.setDirPath(dirPath);
            try {
                floatAudioRecorder.init(context);
            } catch (Exception e) {
                e.printStackTrace();
            }
            return floatAudioRecorder;
        }

    }

}

File: kotlinlibrary/src/main/java/com/bluthlee/library/FloatAudioRecorder.kt
package com.bluthlee.library

import android.content.Context
import com.bluthlee.library.data.BaseBuilder
import com.bluthlee.library.data.DataBinder

/**
 * Created by LC on 2017/8/28.
 */
class FloatAudioRecorder : AudioRecorder() {

    class Builder private constructor() : BaseBuilder() {

        private lateinit var context: Context

        companion object {
            /**
             * 需要指定一个录音按键用作触摸时启动录音
             * */
            fun setContext(context: Context): Builder {
                val builder = Builder()
                builder.context = context
                return builder
            }
        }

        override fun setOnFinishListener(onFinishListener: OnFinishListener): Builder {
            super.setOnFinishListener(onFinishListener)
            return this
        }

        override fun setFileName(fileName: String): Builder {
            super.setFileName(fileName)
            return this
        }

        override fun setDirPath(dirPath: String): Builder {
            super.setDirPath(dirPath)
            return this
        }


        fun build(): FloatAudioRecorder {
            val floatAudioRecorder = FloatAudioRecorder()
            floatAudioRecorder.setStyle(DataBinder.STYLE_FLOAT)
            floatAudioRecorder.setListener(onFinishListener)
            floatAudioRecorder.setFileName(fileName)
            floatAudioRecorder.setDirPath(dirPath)
            floatAudioRecorder.init(context)
            return floatAudioRecorder
        }

    }

}